File: app/src/main/java/com/kalap/contacts/ContactActivity.java
package com.kalap.contacts;

import android.Manifest;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.ActivityCompat;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import com.kalap.contacts.adapters.ViewPagerAdapter;

import java.util.ArrayList;

public class ContactActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener, ViewPager.OnPageChangeListener {

    private static final int CALL_PHONE_REQUEST = 102;
    private Uri data;
    private ViewPager pager;
    private BottomNavigationView navigationView;
    private MenuItem prevMenuItem;

    @Override
    public void onCreate(Bundle savedInstance) {
        super.onCreate(savedInstance);
        data = getIntent().getData();
        setContentView(R.layout.activity_contact);
        if (ContextCompat.checkSelfPermission(ContactActivity.this, Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(ContactActivity.this, new String[]{Manifest.permission.CALL_PHONE}, CALL_PHONE_REQUEST);
        }
        navigationView = (BottomNavigationView) findViewById(R.id.bottom_navigation);
        navigationView.setOnNavigationItemSelectedListener(this);
        pager = (ViewPager) findViewById(R.id.fragment_container);
        ArrayList<Fragment> fragments = new ArrayList<>();
        DialerFragment dialerFragment = DialerFragment.newInstance(data);
        fragments.add(dialerFragment);
        CallLogsFragment callLogsFragment = CallLogsFragment.newInstance();
        fragments.add(callLogsFragment);
        ContactListFragment contactListFragment = ContactListFragment.newInstance();
        fragments.add(contactListFragment);
        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager(),fragments);
        pager.setAdapter(adapter);
        pager.addOnPageChangeListener(this);
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_dial: {
                pager.setCurrentItem(0);
                return true;
            }
            case R.id.action_recent: {
                pager.setCurrentItem(1);
                return true;
            }
            case R.id.action_all: {
                pager.setCurrentItem(2);
                return true;
            }
        }
        return false;
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

    }

    @Override
    public void onPageSelected(int position) {
        if (prevMenuItem != null) {
            prevMenuItem.setChecked(false);
        } else {
            navigationView.getMenu().getItem(0).setChecked(false);
        }
        navigationView.getMenu().getItem(position).setChecked(true);
        prevMenuItem = navigationView.getMenu().getItem(position);
    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }
}

File: app/src/main/java/com/kalap/contacts/ContactActivity.kt
package com.kalap.contacts

import android.Manifest.permission.CALL_PHONE
import android.Manifest.permission.READ_CALL_LOG
import android.Manifest.permission.READ_CONTACTS
import android.content.pm.PackageManager.PERMISSION_GRANTED
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.appcompat.app.AppCompatActivity
import android.view.MenuItem
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.kalap.contacts.calllogs.CallLogsFragment
import com.kalap.contacts.common.longToast
import com.kalap.contacts.common.onPageSelected
import com.kalap.contacts.contact.ContactListFragment
import com.kalap.contacts.dialer.DialerFragment
import kotlinx.android.synthetic.main.activity_contact.*

class ContactActivity : AppCompatActivity(R.layout.activity_contact) {

    private var prevMenuItem: MenuItem? = null

    override fun onCreate(savedInstance: Bundle?) {
        super.onCreate(savedInstance)
        checkPermission()
        bottom_navigation.setOnNavigationItemSelectedListener(::onNavigationItemSelected)
        val dialerFragment = DialerFragment.newInstance(intent.data)
        val callLogsFragment = CallLogsFragment()
        val contactListFragment = ContactListFragment()
        val fragments = arrayListOf<Fragment>() + dialerFragment + callLogsFragment + contactListFragment
        val adapter = ViewPagerAdapter(supportFragmentManager, fragments)
        fragment_container.adapter = adapter
        fragment_container.onPageSelected(::onPageSelected)
    }

    private fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_dial -> {
                fragment_container.currentItem = 0
                return true
            }
            R.id.action_recent -> {
                fragment_container.currentItem = 1
                return true
            }
            R.id.action_all -> {
                fragment_container.currentItem = 2
                return true
            }
        }
        return false
    }

    private fun onPageSelected(position: Int) {
        if (prevMenuItem != null) {
            prevMenuItem!!.isChecked = false
        } else {
            bottom_navigation.menu.getItem(0).isChecked = false
        }
        bottom_navigation.menu.getItem(position).isChecked = true
        prevMenuItem = bottom_navigation.menu.getItem(position)
    }

    private fun checkPermission() {
        val permissions = arrayOf(READ_CONTACTS, CALL_PHONE, READ_CALL_LOG)
        val shouldRequest = permissions.any { checkSelfPermission(it) != PERMISSION_GRANTED }
        if (shouldRequest) {
            requestPermissions(permissions, 101)
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (grantResults.isEmpty() || grantResults.any { it != PERMISSION_GRANTED }) {
            onPermissionRejected()
        }
    }

    private fun onPermissionRejected() {
        longToast(R.string.permission_text)
    }
}

class ViewPagerAdapter(fm: FragmentManager, private val fragments: List<Fragment>) : FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    override fun getItem(position: Int): Fragment = fragments[position]
    override fun getCount(): Int = fragments.size
}
